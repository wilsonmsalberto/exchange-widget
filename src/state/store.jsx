import {
  applyMiddleware,
  combineReducers,
  createStore,
} from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunkMiddleware from 'redux-thunk';

import { rates as ratesReducer } from './rates/reducer';

const reducer = combineReducers({
  rates: ratesReducer,
});

export default createStore(
  reducer,
  undefined,
  composeWithDevTools(applyMiddleware(thunkMiddleware)),
);
