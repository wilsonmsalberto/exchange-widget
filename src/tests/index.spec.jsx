import React from 'react';
import {
  render,
} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { matchers } from 'jest-emotion';

import { Provider } from 'react-redux';

import store from 'state/store';

import App from '..';

expect.extend(matchers);

const getRenderWrapper = (props) => render(
  <Provider store={ store }>
    <App { ...props } />
  </Provider>,
);

describe('App Component', () => {
  describe('Rendering', () => {
    it('should render properly', () => {
      fetch.mockResponse(JSON.stringify({}));
      const wrapper = getRenderWrapper();

      const { asFragment } = wrapper;

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
