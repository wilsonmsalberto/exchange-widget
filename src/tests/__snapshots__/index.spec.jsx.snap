// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Rendering should render properly 1`] = `
<DocumentFragment>
  <main>
    <div
      class="css-19n6d3j"
    >
      <select
        class="css-xu4w98"
      />
      <input
        class="css-1x2g7jt"
        placeholder="0"
        type="text"
        value=""
      />
    </div>
    <div
      class="css-1mv8m18"
    >
      <select
        class="css-xu4w98"
      />
      <input
        class="css-1x2g7jt"
        disabled=""
        placeholder="0"
        type="text"
        value=""
      />
    </div>
    <div
      class="css-1ip6n4i"
    >
      <button
        class="css-15ay18u"
        type="submit"
      >
        Exchange
      </button>
    </div>
  </main>
</DocumentFragment>
`;
